sled-agent/api/src/lib.rs

```diff
@@ -157,6 +159,60 @@ pub trait SledAgentApi {
         rqctx: RequestContext<Self::Context>,
     ) -> Result<HttpResponseOk<Vec<String>>, HttpError>;
 
+    /// List all support bundles within a particular dataset
+    #[endpoint {
+        method = GET,
+        path = "/support-bundles/{zpool_id}/{dataset_id}"
+    }]
+    async fn support_bundle_list(
+        rqctx: RequestContext<Self::Context>,
+        path_params: Path<SupportBundleListPathParam>,
+    ) -> Result<HttpResponseOk<Vec<SupportBundleMetadata>>, HttpError>;
+
+    /// Create a service bundle within a particular dataset
```

Comment:
Confusing to use a name that doesn't match the API path.

```suggestion
    /// Create a support bundle within a particular dataset
```

Reply:
Updated